### PR TITLE
Mingw-w64 on windows doesn't have `setenv`, fix that.

### PR DESCRIPTION
**Pull request by sighingnow**
When compiling with mingw64/mingw-w64-x86_64-gcc on windows I met the following undefined reference error:

```
  CCLD     jq.exe
x86_64-w64-mingw32/bin/ld.exe: ./.libs/libjq.a(builtin.o):builtin.c:(.text+0xc91c): undefined reference to `setenv'
x86_64-w64-mingw32/bin/ld.exe: ./.libs/libjq.a(builtin.o):builtin.c:(.text+0xc949): undefined reference to `setenv'
collect2.exe: error: ld returned 1 exit status
make[2]: *** [Makefile:935: jq.exe] Error 1
make[2]: Leaving directory '/d/Open/jq'
make[1]: *** [Makefile:1104: all-recursive] Error 1
make[1]: Leaving directory '/d/Open/jq'
make: *** [Makefile:775: all] Error 2
```

The appveyor is green, but people do experience the same problem when using mingw-w64 on windows: https://github.com/Alexpux/MINGW-packages/search?p=1&q=setenv&type=&utf8=%E2%9C%93

### CODE REVIEW

File: src/builtin.c [hunk 1]
```diff
@@ -1186,6 +1186,25 @@ static jv tm2jv(struct tm *tm) {
                   jv_number(tm->tm_yday));
 }
 
+#ifndef HAVE_SETENV
```
Comment (nicowilliams) [inline src/builtin.c]: Can we make this:

```
#if defined(WIN32) && !defined(HAVE_SETENV)
```

Comment (sighingnow) [reply src/builtin.c]: Done.

**Comment by sighingnow** [comment]
The one failure on travis shouldn't be related to this PR. All other build tasks run for 13 to 20 minutes, but the last one failed at 1 min 34 sec. Could anyone please look at this and try to re-run the failed task?

**Comment by nicowilliams** [comment]
Merge rebased.  Thanks!

**Pull request merged**